pkg/resources/task.go

```diff
@@ -191,110 +142,73 @@ func Task() *schema.Resource {
 // ReadTask implements schema.ReadFunc.
 func ReadTask(d *schema.ResourceData, meta interface{}) error {
 	db := meta.(*sql.DB)
-	taskID, err := taskIDFromString(d.Id())
-	if err != nil {
-		return err
-	}
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
 
-	database := taskID.DatabaseName
-	schema := taskID.SchemaName
-	name := taskID.TaskName
+	taskId := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
 
-	builder := snowflake.NewTaskBuilder(name, database, schema)
-	q := builder.Show()
-	row := snowflake.QueryRow(db, q)
-	t, err := snowflake.ScanTask(row)
-	if errors.Is(err, sql.ErrNoRows) {
+	task, err := client.Tasks.ShowByID(ctx, taskId)
+	if err != nil {
 		// If not found, mark resource to be removed from state file during apply or refresh
 		log.Printf("[DEBUG] task (%s) not found", d.Id())
 		d.SetId("")
 		return nil
 	}
-	if err != nil {
-		return err
-	}
 
-	if err := d.Set("enabled", t.IsEnabled()); err != nil {
+	if err := d.Set("enabled", task.IsStarted()); err != nil {
```

Comment:
I can't find this in Snowflake documentation so I'm not sure. Should this be IsEnabled()? Or was the previous attribute poorly named?

Reply:
There are two states of the task: "started" and "suspended". Actions are RESUME and SUSPEND.

I think that IsEnabled is a poor name for it, that's why I have renamed it.